### PR TITLE
Added report caller

### PR DESCRIPTION
**Pull request by fqlx**
Added report caller

See issue: https://github.com/sohlich/elogrus/pull/29

### CODE REVIEW

File: hook.go [hunk 1]
```diff
@@ -39,11 +39,13 @@ type ElasticHook struct {
...
-	Message   string        `json:"message"`
-	Data      logrus.Fields `json:"data"`
-	Level     string        `json:"level"`
+	Host      string `json:"host,omitempty"`
+	Timestamp string `json:"@timestamp"`
+	File      string `json:"file,omitempty"`
+	Func      string `json:"func,omitempty"`
+	Message   string `json:"message,omitempty"`
+	Data      logrus.Fields
+	Level     string `json:"level,omitempty"`
```
Comment (denisvmedia) [inline hook.go]: What is the reason you made all the previously existed fields omitempty and removed the struct tag from Data?

Comment (fqlx) [reply hook.go]: Hi, I cherry picked this fix from https://github.com/sohlich/elogrus/pull/29. I'm not the author.

I've noticed this fixes two issues:
1) `Data` field was not being sent to ES.  It is broken currently. That MR fixed sending the `Data` field.
2) Adds `File` and `Func` when caller reporting is enabled.

Regarding the optional fields:
- `File` and `Func` are optional fields based on caller reporting being enabled.
- I'm not sure what the reasoning for `Host` and `Level` to now be optional fields.  I'm guessing to allow a user to pass `nil` when creating a hook.  
e.g.
```
  hook, err := elogrus.NewAsyncElasticHook(
    client,
    host, // Can be nil
    config.LogLevel, // Can be nil
    config.LogSource,
  )
  ```

Comment (denisvmedia) [reply hook.go]: I see, thanks for clarifications. Can you please at least add `json:"data,omitempty"` to the Data field (for consistency). And I will merge the PR.

File: hook.go [hunk 2]
```diff
@@ -39,11 +39,13 @@ type ElasticHook struct {
...
-	Message   string        `json:"message"`
-	Data      logrus.Fields `json:"data"`
-	Level     string        `json:"level"`
+	Host      string `json:"host,omitempty"`
+	Timestamp string `json:"@timestamp"`
+	File      string `json:"file,omitempty"`
+	Func      string `json:"func,omitempty"`
+	Message   string `json:"message,omitempty"`
+	Data      logrus.Fields `json:"data,omitempty"`
+	Level     string `json:"level,omitempty"`
```
Comment (denisvmedia) [inline hook.go]: Can you please also run go fmt on this file? The alignment of the fields is not done according to `go fmt`.

Comment (fqlx) [reply hook.go]: Done

File: hook.go (hunk 1 above)
Comment (fqlx) [reply hook.go]: Done

**Comment by denisvmedia** [comment]
Thank you!

**Comment by fqlx** [comment]
Are you able to update the release on `gopkg.in`? So I can pull these changes

**Pull request merged**